lib/vdc/BOVCollection.cpp

```diff
@@ -216,12 +218,14 @@ int BOVCollection::_parseHeader(std::ifstream &header)
 int BOVCollection::_validateParsedValues()
 {
     // Validate grid dimensions
-    if (_tmpGridSize[0] < 1 || _tmpGridSize[1] < 1 || _tmpGridSize[2] < 1)
+    if (_tmpGridSize[0] < 2 || _tmpGridSize[1] < 2 || _tmpGridSize[2] < 2)
         return _invalidDimensionError(GRID_SIZE_TOKEN);
-    else if (_tmpGridSize != _gridSize && _gridSizeAssigned == true)
+    else if ((_tmpGridSize[0] != _gridSize[0] || _tmpGridSize[1] != _gridSize[1] || _tmpGridSize[2] != _gridSize[2]) && _gridSizeAssigned == true)
```

Comment:
Can you use parenthesis to group logical operations together so the overall logic is clearer? 

In other words, does a compiler treat `(a || b || c && d)` the same as you think it would?

Reply:
I blame this on clang-format.  They way I originally wrote it was as so:

```
    else if (( _tmpGridSize[0] != _gridSize[0] 
                || _tmpGridSize[1] != _gridSize[1] 
                || _tmpGridSize[2] != _gridSize[2]) 
              && _gridSizeAssigned == true)

```

Yes, the logic does what's intended, which is to make sure all _tmpGridSize values match those of _gridSize; and then that gridSize has also been assigned.  If it hasn't been assigned, then _tmpGridSize is allowed to differ from the default value of _gridSize.

Reply:
The look adequately and unambiguously  parenthesized to me. Not sure what @shaomeng is asking for.

Reply:
My bad, I missed the inner parenthesis. Now I see the logic very clear. Sorry guys. (Human reviewing aren't reliable sometimes...)